reactive-streams/src/main/scala/fs2/interop/reactivestreams/StreamSubscriber.scala

```diff
@@ -199,7 +189,6 @@ object StreamSubscriber {
             chunk <- q.dequeueChunk1(maxDemand)
             _ <- ref.get flatMap {
               case Receiving(s) => F.delay(s.request(chunk.size))
-              case Idle(s) => F.delay(s.request(chunk.size))
               case _ => q.enqueue1(None.asRight)
```

Comment:
Can we simplify this to `F.unit`? We don't need to worry about terminating the queue here as if we're in a terminal state it will have already happened when handling the terminal input.